### DISCUSSION TITLE
Estimating state of the battery with PyBaMM

### DISCUSSION THREAD

**Discussion by caz3im** [pybamm-team/PyBaMM#3822] [General]
Hi, I just started learning PyBaMM couple of days ago and have a question regarding state estimation (SOC, SOH, SOP and so on).
What would be the simplest way to estimate for example SOC or SOH of a battery that has been discharged with some arbitrary current signal.

Are there some variables from a DFN model, or some other model, that can be used to simply calculate a state (SOC, SOH) at any given time? For example integrating the Discharged Capacity variable should give me SOC, if I know the inital SOC and the capacity of the battery at 100% SOC, right?
Do you think that there are some models that would be better at this?

**Comment by valentinsulzer** [comment]
PyBaMM is not built for state estimation, but the approach you describe for calculating SOC will work.
